extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -105,7 +105,14 @@ export class AmpIframe extends AMP.BaseElement {
         !((' ' + sandbox + ' ').match(/\s+allow-same-origin\s+/i)),
         'allow-same-origin is not allowed with the srcdoc attribute %s.',
         this.element);
-    return 'data:text/html;charset=utf-8;base64,' + btoa(srcdoc);
+
+    // srcdoc is a DOMString and requires special handling for base64 encoding.
+    // See https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+    const encodedSrcdoc = btoa(encodeURIComponent(srcdoc)
```

Comment:
`utf8Encode` is async indeed. How about we change `base64Encode` to support unicode encoding:
```js
export function base64Encode(str) {
  return btoa(unescape(encodeURIComponent(str)));
}
export function base64Decode(str) {
  return decodeURIComponent(escape(atob(str)));
}
```
And `base64Encode` can be called in `amp-iframe`.